### ISSUE TITLE
Fatal Exception: java.lang.SecurityException !@Too many alarms (500)

### ISSUE THREAD

**Issue by vhontar** [OneSignal/OneSignal-Android-SDK#887]
## Goals
I want to avoid background crashes which happen to the users all the time.
## Expected Results

https://github.com/OneSignal/OneSignal-Android-SDK/blob/master/OneSignalSDK/onesignal/src/main/java/com/onesignal/NotificationRestorer.java#L259

Could you please at least put alarm set to try/catch to avoid crashes in the background
```
try {
 alarm.set(AlarmManager.RTC, scheduleTime, pendingIntent);
} catch (SecurityException e) {}
```
## Actual Results

```
Caused by java.lang.SecurityException
!@Too many alarms (500) registered from uid 10011
android.os.Parcel.readException (Parcel.java:1693)
android.app.AlarmManager.set (AlarmManager.java:391)
com.onesignal.NotificationRestorer.startDelayedRestoreTaskFromReceiver (NotificationRestorer.java:259)
com.onesignal.BootUpReceiver.onReceive (BootUpReceiver.java:38)
android.app.ActivityThread.handleReceiver (ActivityThread.java:3362)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1386)
```
## Steps to Reproduce

If you've set up already more than 500 alarms by the app. Some devices, especially Samsung (android 7) has some security measures to forbid to create more than 500 alarms.

Is it possible to add this try/catch code to avoid such crashes in the future?

**Comment by rgomezp** [comment]
Thanks for bringing this up. Could you please provide further info such as the environment you reproduced this in?

**Comment by janbolat** [comment]
I faced same problem in the past when using alarms.
The fix is simple, just change it FLAG_UPDATE_CURRENT.

**Comment by vhontar** [comment]
The problem is that I have to set alarms for different local notifications for different time. So request code is different. That's why they are all different. FLAG_UPDATE_CURRENT as I understand right doesn't help in that case
```
 private fun cancelAlarm(myBookingNotification: MyBookingNotification) {
        val notificationIntent = getNotificationIntent(myBookingNotification)
        val pendingIntent = PendingIntent.getBroadcast(context, myBookingNotification.uniqueNotificationId, notificationIntent, PendingIntent.FLAG_UPDATE_CURRENT)
        val alarmManager = context.getSystemService(Context.ALARM_SERVICE) as AlarmManager
        if (pendingIntent != null) {
            alarmManager.cancel(pendingIntent)
        }
    }
```

**Comment by Ariandr** [comment]
Hi @rgomezp 
The similar bug was already fixed in this library (here https://github.com/OneSignal/OneSignal-Android-SDK/commit/b26fa8d1d5922151bb1e7616200c22ddb16c5888). It was related to this bug https://github.com/OneSignal/OneSignal-Android-SDK/issues/168

You need to change `FLAG_CANCEL_CURRENT` to `FLAG_UPDATE_CURRENT` here https://github.com/OneSignal/OneSignal-Android-SDK/blob/master/OneSignalSDK/onesignal/src/main/java/com/onesignal/NotificationRestorer.java (line 255)

**Comment by jkasten2** [comment]
This is fixed in PR #910, it will be included in the next release!
